opm/grid/cpgpreprocess/preprocess.c

```diff
@@ -716,29 +717,178 @@ vert_size(const struct grdecl *in,
 
 
 /* ---------------------------------------------------------------------- */
-static int
-is_lefthanded(const struct grdecl *in, const double sign)
+static void
+vertex_coord(const struct grdecl *in,
+             const size_t         pillar,
+             const double         z,
+             double              *coord)
+/* ---------------------------------------------------------------------- */
+{
+    const double *top = &in->coord[6*pillar + 0];
+    const double *bot = &in->coord[6*pillar + 3]; /* == top + 3 */
+
+    const int coincide = fabs(top[2] - bot[2]) < 1.0e-6;
+
+    const double t = coincide
+        ? 0.0 /* coincide => vertical */
+        : (z - top[2]) / (bot[2] - top[2]); /* straight line */
+
+    coord[0] = (1.0 - t)*top[0] + t*bot[0];
+    coord[1] = (1.0 - t)*top[1] + t*bot[1];
+    coord[2] = z;
+}
+
+
+static void
+bounding_box_cross_axes(const double *origin,
+                        const double *i_axis,
+                        const double *j_axis,
+                        double       *cross)
+{
+    cross[0] = (i_axis[1] - origin[1]) * (j_axis[2] - origin[2])
```

Comment:
Reuse cross from geometry.c?

Reply:
> Reuse cross from geometry.c?

Possibly.  The one downside to doing that is it'd make the function callable in the general library since we don't control symbol visibility.  I suppose we could extract the function as `cpg_cross_product()` or something along those lines.